lib/dk_buttons_and_leds/dk_buttons_and_leds.c

```diff
@@ -22,32 +22,41 @@ struct gpio_pin {
 };
 
 static const struct gpio_pin button_pins[] = {
-#ifdef DT_ALIAS_SW0_GPIOS_PIN
-	{DT_ALIAS_SW0_GPIOS_CONTROLLER, DT_ALIAS_SW0_GPIOS_PIN},
+#if DT_NODE_EXISTS(DT_GPIO_PIN(DT_ALIAS(sw0), gpios))
```

Comment:
... yes, definitely

Reply:
I believe this should be DT_NODE_EXISTS(DT_ALIAS(sw0)). Similar comments elsewhere in the PR.

DT_GPIO_PIN(...) will return a pin number, so it is not a devicetree node identifier. The argument to DT_NODE_EXISTS must be a node identifier. In this case, DT_ALIAS(sw0) is a node identifier for the node the /aliases property sw0 points at.

Since this is our DK buttons and LEDs library, I think we can just check for the existence of the alias at all, rather than making sure that its gpios property has a pin cell, since we know all our boards behave properly.

If you really want to check for the existence of a pin cell (which I don't recommend), I would use DT_PHA_HAS_CELL(DT_ALIAS(sw0), gpios, pin) instead.

Reply:
Changed to DT_NODE_EXISTS(DT_ALIAS(sw0)). Thanks.